### PR TITLE
Refactor: block router

### PR DESCRIPTION
**Pull request by luixo**

### CODE REVIEW

File: backend/src/router/block.ts [hunk 1]
```diff
@@ -3,46 +3,183 @@ import { z } from "zod";
...
+import { indexerDatabase } from "../database/databases";
+import { count, div, sum } from "../database/utils";
+import { millisecondsToNanoseconds } from "../utils/bigint";
 
 export const router = trpc
   .router<Context>()
-  .query("nearcore-final-block", {
-    resolve: () => {
-      return nearApi.sendJsonRpc("block", { finality: "final" });
+  .query("final", {
```
Comment (luixo) [inline backend/src/router/block.ts:13]: That looks for me like a more sensible way - as we use `nearcore-final-block` for just getting the block's height - to use it for both calls and resolve with just actual height.

File: backend/src/router/block.ts [hunk 2]
```diff
@@ -3,46 +3,183 @@ import { z } from "zod";
...
+        hash: selection.hash!,
+        // TODO: Discover how to get rid of non-null type assertion
+        height: parseInt(selection.height!),
+        timestamp: parseInt(selection.timestamp),
+        // TODO: Discover how to get rid of non-null type assertion
+        prevHash: selection.prevHash!,
+        // TODO: Discover how to get rid of non-null type assertion
+        gasPrice: selection.gasPrice!,
+        // TODO: Discover how to get rid of non-null type assertion
+        // TODO: Remove totalSupply whenever `epochStartBlock` will be served
```
Comment (luixo) [inline backend/src/router/block.ts:103]: Thats a point I should not forget

File: frontend/src/components/utils/Search.tsx [hunk 3]
```diff
@@ -186,19 +186,9 @@ const Search: React.FC<Props> = React.memo(({ dashboard }) => {
 
       const cleanedSearchValue = value.replace(/\s/g, "");
 
-      let blockPromise: Promise<string | null>;
-      const maybeBlockHeight = cleanedSearchValue.replace(/[,]/g, "");
```
Comment (luixo) [inline frontend/src/components/utils/Search.tsx:190]: No need for complicated logic on client - just check if it's a number server-side and return hash.

File: frontend/src/pages/api/metrics/indexer.ts [hunk 4]
```diff
@@ -7,9 +7,11 @@ const handler: NextApiHandler = async (req, res) => {
   try {
     const networkName = getNearNetworkName(req.query, req.headers.host);
     const trpcClient = getTrpcClient(networkName);
-    const rpcFinalBlock = await trpcClient.query("nearcore-final-block");
-    const indexerFinalBlock = await trpcClient.query("blocks-list", {
-      limit: 1,
+    const rpcFinalBlock = await trpcClient.query("block.final", {
```
Comment (luixo) [inline frontend/src/pages/api/metrics/indexer.ts:10]: Isn't that more comprehendible?

Comment (shelegdmitriy) [reply frontend/src/pages/api/metrics/indexer.ts:10]: That's much better!

**Pull request merged**